Not-ending learning

I think I am not the only how experienced when you start a "General" learning and it never ends

![image](https://user-images.githubusercontent.com/1022757/48978462-e4ec2800-f0bc-11e8-8a2a-985bbf849e48.png)

```
2018-12-11 12:41:23,076 [Analytics] [ERROR]  handle_analytic_task exception: 'Traceback (most recent call last):
  File "bin/../analytics/analytic_unit_manager.py", line 77, in handle_analytic_task
    result_payload = await self.__handle_analytic_task(task)
  File "bin/../analytics/analytic_unit_manager.py", line 71, in __handle_analytic_task
    return await worker.do_detect(data, payload['cache'])
  File "bin/../analytics/analytic_unit_worker.py", line 34, in do_detect
    return self._detector.detect(data, cache)
  File "bin/../analytics/detectors/pattern_detector.py", line 50, in detect
    detected = self.model.detect(dataframe, cache)
  File "bin/../analytics/models/model.py", line 47, in detect
    ) for x in result]
  File "bin/../analytics/models/model.py", line 47, in <listcomp>
    ) for x in result]
  File "/usr/local/lib/python3.6/site-packages/pandas/core/series.py", line 601, in __getitem__
    result = self.index.get_value(self, key)
  File "/usr/local/lib/python3.6/site-packages/pandas/core/indexes/base.py", line 2477, in get_value
    tz=getattr(series.dtype, 'tz', None))
  File "pandas/_libs/index.pyx", line 98, in pandas._libs.index.IndexEngine.get_value (pandas/_libs/index.c:4404)
  File "pandas/_libs/index.pyx", line 106, in pandas._libs.index.IndexEngine.get_value (pandas/_libs/index.c:4087)
  File "pandas/_libs/index.pyx", line 154, in pandas._libs.index.IndexEngine.get_loc (pandas/_libs/index.c:5126)
  File "pandas/_libs/hashtable_class_helper.pxi", line 759, in pandas._libs.hashtable.Int64HashTable.get_item (pandas/_libs/hashtable.c:14031)
  File "pandas/_libs/hashtable_class_helper.pxi", line 765, in pandas._libs.hashtable.Int64HashTable.get_item (pandas/_libs/hashtable.c:13975)
KeyError: 2794
'
```

--- a/analytics/requirements.txt
+++ b/analytics/requirements.txt
@@ -10,7 +10,7 @@ patsy==0.5.0
 pefile==2017.11.5
 python-dateutil==2.7.3
 pytz==2018.4
-pyzmq==17.0.0
+pyzmq==18.0.1
 scikit-learn==0.19.1
 scipy==1.1.0
 seglearn==0.1.6

